### ISSUE TITLE
Unexpected error when setting loss_weights for multi-output model

### ISSUE THREAD

**Issue by davechallis** [fchollet/keras#3212]
According to the functional API documentation at http://keras.io/getting-started/functional-api-guide/#multi-input-and-multi-output-models , I can pass a single list for the `loss_weights` argument to use it on all output.

However when I tried this (on a model with 6 inputs and 6 outputs), e.g.:

    model = Model(input=all_inputs, output=all_output)
    model.compile(optimizer='adam',
                  loss='categorical_crossentropy',
                  metrics=['accuracy'],
                  loss_weights=[1.0, 0.1])
 
I get the error:

```
/Users/dsc/miniconda3/envs/tf/lib/python3.5/site-packages/keras/engine/training.py in compile(self, optimizer, loss, metrics, loss_weights, sample_weight_mode, **kwargs)
    486                                 'The model has ' + str(len(self.outputs)) +
    487                                 ' outputs, but you passed loss_weights=' +
--> 488                                 str(loss))
    489             loss_weights_list = loss_weights
    490         else:

Exception: When passing a list as loss_weights, it should have one entry per model outputs. The model has 6 outputs, but you passed loss_weights=categorical_crossentropy
```

It seems to want separate loss weights per output, and it also seems to be picking up the `loss` argument as the `loss_weights` argument for some reason (though perhaps that's just a mistake in error reporting?).

**Comment by davechallis** [comment]
Hmm.. looks like I misunderstood loss_weights (it's per output, not per class from what I can see). The other issue is just related to error reporting, will fix and submit a pull request.
